### PR TITLE
MOTECH-2740: Improve input-map directive display

### PR DESCRIPTION
**Pull request by afijal**
Some changes to the input-map directive css, to make it look prettier. Displays pretty well and as requested on mobile devices.

### CODE REVIEW

**Comment by nickdotreid** [comment]
@afijal -- I see this is the same as PR #32 (and I bugged you about mixing PRs there too, so I'll try not to sound like a broken record)

Do you want to separate your PRs, or deal with them in one PR? If you want to just deal with one PR, close this PR — otherwise rebase in both places and let me know what I can merge #32 and we will deal with MOTECH-2470 here

thanks!

**Comment by afijal** [comment]
@nickdotreid I used `display:flex` according to your suggestions, and it looks much better now, both in code and UI, thank you for your advice, I forgot about existence of this display type. Looks as requested on mobile devices too, but add button is on the right side of value input now, not below it, however, it is inlined with delete buttons now.
Also, regarding to the ticket, I changed spacing between key-value pairs "as to appear these items have more in common", but I have mixed feelings about that. I guess maybe we should use bigger font in Labels (like "Motech event parameters").
Let me know if it looks ok for you now, Cheers.

**Comment by nickdotreid** [comment]
This looks great! Minor tweak and a bug that I'm not sure if its related

Can we add a little space above the key-value pair grouping when its collapsed on a small screen 

and make the add button go full width -- allowing the text to show?
image

Other random error -- when I press enter in any of the text boxes then a single entry from the input map disappears... which is really odd behavior

I don't think we want to hijack the enter key when the input map has focus, but I'd like to not loose enteries if possible -- I'm happy to file this as another bug as its extremely low priority...

**Comment by afijal** [comment]
@nickdotreid 
I added requested margin, but it is showed both on mobile and desktop, is it ok for You, also let me know if its size fits (screenshot below). I don't really understand what do You mean by

> and make the add button go full width -- allowing the text to show?
I guess You didn't mean that:
zrzut ekranu z 2016-07-27 13-00-15

**Comment by nickdotreid** [comment]
hahahaha... thats pretty stupid looking -- yes I didn't mean that (kinda)

I meant to do that in a max-width media query, so only if $screen-sm (I think that is the variable) we get a full width clickable add button.

I think the margins are great

**Comment by motech-gerrit** [comment]
Can one of the admins verify this patch?

**Comment by sebbrudzinski** [comment]
test this please

File: src/common/input-map/input-map.scss [hunk 1]
```diff
@@ -2,18 +2,18 @@
...
 .input-map .form-group{
     padding-left: 0px;
+    margin-bottom: 0.5em;
     >*{
         display: inline-block;
-    }
-    label, input{
-        width: $sidebar-width;
+        &:last-child{
+            margin-left: 15px;
```
Comment (jredlarski) [inline src/common/input-map/input-map.scss]: @afijal Use `$space-size` which in fact represents 15px instead of using magic numbers

**Comment by jredlarski** [comment]
@nickdotreid @afijal The add button is behaving as requested now.

After changes in common .scss files the login modal changed it appearance from:
motech-sign-in-modal-before

to this: 

motech-sign-in-modal-after

Apart from login modal this issue appears in any modal which contains a label with an input.

I believe that there are two ways of fixing it:
**(1)** Edit common .scss files so do not modify the look of the login modal and style only the motech input-map.
**(2)** Remove labels from login modal and add `placeholder` for both fields which would change the look of the login modal to:
motech-sign-in-modal-placeholders
which is even more similar to the old one used in the Motech Core. But such modification would have take place in every modal affected...

I would rather suggest going for the **(1)** option. @nickdotreid I would love to hear what you think about it...

**Comment by nickdotreid** [comment]
@jredlarski I'm going to state a huge opinion of mine — **I hate placeholder text.** I think it makes it harder for users to correct their errors, and isn't accessible. I know that placeholder text is used all over MOTECH, and that it makes forms look prettier. I just still think its bad form. I'm willing to live with it in complex UIs that are already implemented...

...moving on...

Yes, option (1) is the way to go — the problem probably has to do with more specific CSS that should be wrapped in a media query.

I'll try to do some splunking, and see if I can give a more specific example

**Comment by jredlarski** [comment]
@nickdotreid Yeah, sure! That was just an idea for an alternative solution to a problem. On second thought I must admit that I agree with your point of view and we also agreed that we are not copying the Motech Core styling directly.

@afijal Let's try to figure out what is messing the look of the forms in modal windows

**Comment by jredlarski** [comment]
test this please

**Comment by jredlarski** [comment]
test this please

**Pull request merged**